vispy/util/osmesa_gl.py

```diff
@@ -0,0 +1,20 @@
+import os
+from . import logger
+
+
+def fix_osmesa_gl_lib():
+    """
+    When using OSMesa, the gl functions (from libGL) are included
+    in libOSMesa.so. This function modifies the VISPY_GL_LIB env variable
+    so gl2 picks up libOSMesa.so as the OpenGL library.
+
+    This modification must be done before vispy.gloo is imported for the
+    first time.
+    """
+    if 'VISPY_GL_LIB' in os.environ:
+        logger.warning('VISPY_GL_LIB is ignored when using OSMesa. Use '
+                       'OSMESA_LIBRARY instead.')
+    if 'OSMESA_LIBRARY' in os.environ:
+        os.environ['VISPY_GL_LIB'] = os.environ['OSMESA_LIBRARY']
```

Comment:
can just be `os.environ['VISPY_GL_LIB'] = os.getenv('OSMESA_LIBRARY', 'libOSMesa.so')`